src/main/java/org/spongepowered/common/mixin/core/data/types/MixinBlockPistonExtensionEnumPistonType.java

```diff
@@ -38,12 +43,34 @@
 
     @Shadow public abstract String getName();
 
+    private Translation translation;
+
+    @Inject(method = "<init>", at = @At("RETURN") )
+    public void onConstructed(String internalName, int internalOrdinal, String name, CallbackInfo ci) {
+        final String translationId;
+        switch (name) {
```

Comment:
Yeah, I can't stress this enough, don't use a switch for strings even. For something so simple, just make an `if...else...` statement.